### PR TITLE
Html page redesign

### PR DESCRIPTION
**Pull request by TillK17**
## Context

[AI/ai-sdk-java-backlog#I128](https://github.tools.sap/orgs/AI/projects/5/views/1?pane=issue&itemId=55773)

I have redesigned the Frontend of our sample app, including endpoint descriptions (based on JavaDoc), JSON rendering ([GitHub Repo of tool](https://github.com/abodelot/jquery.json-viewer)), toggling between full json view or only showing response message (for calls to LLM with message output), streaming functionality and minimal error handling. 

### Feature scope:
 
- [x] Prettify the page to look like an actual modern web page
    - Claude is your friend for this 😉 
- [x] Add descriptions for endpoints (add links, what is OpenAI, what is orchestration service, "here we try to trigger a content filter expecting the following result")
- [x] Render response text content instead of plain JSON response

## Definition of Done

- [x] Functionality scope stated & covered
- [ ] ~~Tests cover the scope above~~
- [ ] ~~Error handling created / updated & covered by the tests above~~
- [ ] ~~Aligned changes with the JavaScript SDK~~
- [ ] ~~Documentation updated~~
- [ ] ~~Release notes updated~~

### CODE REVIEW

**Comment by TillK17** [comment]
- I would especially be interested in feedback regarding the explanation of the different API parts as many of you are most likely able to explain them better than me
- as there is no final decision on our new logo yet, I've agreed with @jjtang1985 to add this later

**Comment by TillK17** [comment]
- last commit is adapted to new backend of sample app with differentiation of JSON or content response by header field in request
- global toggle to switch between JSON rendering or plain text output
- not tested with new backend

**Comment by Jonas-Isr** [comment]
I merged the current state into this branch and fixed the behavior regarding the new backend. I also added minor improvements to the code.

**Review by newtork — APPROVED** [review]

File: sample-code/spring-app/src/main/resources/static/index.html [hunk 1]
```diff
@@ -21,77 +25,605 @@
...
+                                    Chat request to OpenAI through the Orchestration service with a simple prompt.
+                                </div>
+                            </div>
+                        </li>
+                        <li class="list-group-item">
+                            <div class="info-tooltip">
+                                <a class="link-offset-2-hover link-underline link-underline-opacity-0 link-underline-opacity-75-hover endpoint"
+                                   href="/orchestration/streamChatCompletion"><code>/orchestration/streamChatCompletion</code></a>
+                                <div class="tooltip-content">
+                                    Chat request to OpenAI through the Orchestration service with a simple prompt.
```
[the remark this answers is not in the document]
Comment (TillK17) [reply sample-code/spring-app/src/main/resources/static/index.html:266]: Should we open a new ticket for changing this in the JavaDoc as well? Or should I just change it with another commit in this PR?

[the remark this answers is not in the document]
Comment (rpanackal) [reply sample-code/spring-app/src/main/resources/static/index.html]: Thank you for changing it. They were just leftovers from some old ticket that was missed at the time.

**Review by rpanackal — APPROVED** [review]
Looks good to me.

**Pull request merged**